### DISCUSSION TITLE
Distortion of custom shape if text is too long

### DISCUSSION THREAD

**Discussion by alexis-n** [clientIO/joint#2395] [Q&A] [answered]
### Introduction

Hello JointJS Team,

I'm currently using custom shapes in the JointJS library.  However, I am having issues with custom shapes and the text associated with these shapes. Indeed, as you can see in the image, if the text of the shape is wider than the shape itself, the shape is distorted.

<img width="472" alt="image" src="https://github.com/clientIO/joint/assets/2537671/401913e7-4dc1-4461-b5c0-84076ba86d59">

The problem does not occur on standard shapes, which makes me say that the problem comes from the SVG of my shape.

`<g @selector="scalable">
        <path d="M44.19,2.03L1.25,79.89c-.74,1.33,.23,2.97,1.75,2.97H88.87c1.52,0,2.49-1.63,1.75-2.97L47.69,2.03c-.76-1.38-2.74-1.38-3.5,0Z" fill="#eff5fc" stroke="#131232" stroke-linecap="round" stroke-linejoin="round" stroke-width="2"/>
        <g>
          <g>
            <polyline points="31.25 70.86 46.52 70.86 60.62 70.86" fill="#eff5fc" stroke="#131232" stroke-linecap="round" stroke-linejoin="round" stroke-width="2"/>
            <polyline points="31.25 41.78 45.94 41.78 60.62 41.78" fill="#eff5fc" stroke="#131232" stroke-linecap="round" stroke-linejoin="round" stroke-width="2"/>
          </g>
          <line x1="45.94" y1="41.78" x2="45.94" y2="70.86" fill="#eff5fc" stroke="#131232" stroke-linecap="round" stroke-linejoin="round" stroke-width="2"/>
        </g>
        <text @selector="text1" text-anchor="middle" transform="translate(45.04 97.39)" font-family="ArialMT, Arial" font-size="9"><tspan x="0" y="0">i</tspan></text>
      </g>`

The text is set like this in my shape : 
`<text @selector="text1" text-anchor="middle" transform="translate(45.04 97.39)" font-family="ArialMT, Arial" font-size="9"><tspan x="0" y="0">i</tspan></text>`

Do you have any idea on how to solve this problem ?

### Does your question relate to JointJS or JointJS+. Select both if applicable.

JointJS, JointJS+

**Comment by kumilingus** [comment, marked as answer]
Perhaps you should move the text outside of the `scalable` group (the one that overlaps, not the `i` character).
